jwst/outlier_detection/coron.py

```diff
@@ -0,0 +1,84 @@
+"""
+Main class for performing outlier detection.
+
+This is the controlling routine for the outlier detection process.
+It loads and sets the various input data and parameters needed by
+the various functions and then controls the operation of this process
+through all the steps used for the detection.
+
+Notes
+-----
+This routine performs the following operations::
+
+  1. Extracts parameter settings from input model and merges
+     them with any user-provided values
+  2. Resamples all input images into grouped observation mosaics.
+  3. Creates a median image from all grouped observation mosaics.
+  4. Blot median image to match each original input image.
+  5. Perform statistical comparison between blotted image and original
+     image to identify outliers.
+  6. Updates input data model DQ arrays with mask of detected outliers.
+
+"""
+
+
+import logging
+
+import numpy as np
+
+from stdatamodels.jwst import datamodels
+
+from jwst.resample.resample_utils import build_mask
+
+from .utils import create_cube_median, flag_model_crs
+from ._fileio import save_median
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+__all__ = ["detect_outliers"]
+
+
+def detect_outliers(
+    input_model,
+    save_intermediate_results,
+    good_bits,
+    maskpt,
+    snr,
+    asn_id,
+    make_output_path,
+):
+    """Flag outlier pixels in DQ of input images."""
```

Comment:
These arguments (almost) all come from the main step spec. How about a short docstring stating something like?
```
Refer to `OutlierDetectionStep.spec` for documentation of these arguments.
```
This would always be up to date (wrong documentation is probably less helpful than no documentation).

When this step is split up each will have it's own spec which will document these arguments and these functions will no longer be needed.

Reply:
Fair enough RE each having their own spec in the near future.  I think that would fix the problem

Reply:
I added "stub" docstrings like that mentioned above:
7440ce5363540cec1fcaab8bd9bd82d395c3d9cd